src/garage/tf/core/parameter.py

```diff
@@ -3,43 +3,31 @@
 import tensorflow as tf
 
 
-def parameter(input_var,
-              length,
-              batch_dim=None,
+def parameter(length,
```

Comment:
at this point this is just a wrapped call to `tf.Variable`, which is not very useful.

I know we've gone back-and-forth on this. maybe the better solution is to leave `parameter` as-is and just make a special helper function for recurrent policies? it seems that the recurrent case has 2 big differences:

1. you need to broadcast differently
1. you need to reuse the variable

perhaps you can just have `recurrent_parameter()` which handles the broadcasts automatically and returns two outputs. this also helps us encode the proper batch positions for recurrent networks.

Reply:
so `parameter()` will create a new parameter and broadcast, assuming the 0th dim is batch dim and returned the broadcasted parameter while `recurrent_parameter()` will create a new parameter and broadcast, assuming the 0th and 1st dim are batch dims, and return both the broadcasted parameter and the original variable?

Does it sound good?

Reply:
sounds good

Reply:
I found that eventually we always want to broadcast again for step inputs using the original parameter, in recurrent case. Why not do it both at the same time inside `recurrent_parameter`?

Reply:
that sounds good to me.